core/src/test/scala/zio/config/NestedConfigTest.scala

```diff
@@ -67,15 +68,16 @@ object NestedConfigTestUtils {
   final case class TestParams(value: AppConfig) {
 
     val config: ConfigDescriptor[AppConfig] = {
-      val credentials  = (string("user") |@| string("password"))(Credentials.apply, Credentials.unapply)
-      val dbConnection = (string("host") |@| int("port"))(DbConnection.apply, DbConnection.unapply)
+      val credentials  = (string("user") |@| string("password")).to[Credentials]
+      val dbConnection = (string("host") |@| int("port")).to[DbConnection]
 
       val database =
-        (string("dburl")(DbUrl.apply, DbUrl.unapply)
+        (string("dburl")
+          .to[DbUrl]
           .orElseEither(nested("connection")(dbConnection)) |@|
-          nested("credentials")(credentials).optional)(Database.apply, Database.unapply)
+          nested("credentials")(credentials).optional).to[Database]
 
-      (nested("database")(database) |@| double("pricing"))(AppConfig, AppConfig.unapply)
+      (nested("database")(database) |@| double("pricing")).to[AppConfig]
```

Comment:
No a PR comment since it is in very early state, but curious:
I am just searching for `.to[CaseObject]`. Is there one anywhere?

Reply:
Hm I think that would be simple to support in the scala2 macro but I have no idea how to do it in scala3, but worths a try